src/QsCompiler/Transformations/Attributes.cs

```diff
@@ -42,37 +30,66 @@ private Attributes(IEnumerable<(QsDeclarationAttribute, CallablePredicate)> attr
             SyntaxGenerator.StringLiteral(NonNullable<string>.New(target), ImmutableArray<TypedExpression>.Empty);
 
         public static QsCompilation AddToCallables(QsCompilation compilation, QsDeclarationAttribute attribute, CallablePredicate predicate = null) =>
-            new Attributes(new[] { (attribute, predicate) }).Apply(compilation);
+            new AddAttributes(new[] { (attribute, predicate) }).Apply(compilation);
 
         public static QsCompilation AddToCallables(QsCompilation compilation, params (QsDeclarationAttribute, CallablePredicate)[] attributes) =>
-            new Attributes(attributes).Apply(compilation);
+            new AddAttributes(attributes).Apply(compilation);
 
         public static QsCompilation AddToCallables(QsCompilation compilation, params QsDeclarationAttribute[] attributes) =>
-            new Attributes(attributes.Select(att => (att, (CallablePredicate)null))).Apply(compilation);
+            new AddAttributes(attributes.Select(att => (att, (CallablePredicate)null))).Apply(compilation);
 
         public static QsNamespace AddToCallables(QsNamespace ns, QsDeclarationAttribute attribute, CallablePredicate predicate = null) =>
-            new Attributes(new[] { (attribute, predicate) }).Namespaces.OnNamespace(ns);
+            new AddAttributes(new[] { (attribute, predicate) }).Namespaces.OnNamespace(ns);
 
         public static QsNamespace AddToCallables(QsNamespace ns, params (QsDeclarationAttribute, CallablePredicate)[] attributes) =>
-            new Attributes(attributes).Namespaces.OnNamespace(ns);
+            new AddAttributes(attributes).Namespaces.OnNamespace(ns);
 
         public static QsNamespace AddToCallables(QsNamespace ns, params QsDeclarationAttribute[] attributes) =>
-            new Attributes(attributes.Select(att => (att, (CallablePredicate)null))).Namespaces.OnNamespace(ns);
-
+            new AddAttributes(attributes.Select(att => (att, (CallablePredicate)null))).Namespaces.OnNamespace(ns);
 
-        // helper classes
 
-        private class NamespaceTransformation
-        : Core.NamespaceTransformation<AttributeSelection>
+        /// <summary>
+        /// Transformation to add attributes to an existing compilation.
+        /// </summary>
+        private class AddAttributes
```

Comment:
This is a more descriptive name than `Attributes` for the top-level class. Since this class is private it is less important that this has a descriptive name than for the public class. I would suggest swapping the names for this and the top-level class.